### PR TITLE
MBS-13361: Upgrade the required version of PostgreSQL to 16

### PR DESCRIPTION
**Pull request by mwiencek**
# Problem

As explained in MBS-13361 and the [schema change announcement blog post](https://blog.metabrainz.org/2024/03/12/schema-change-release-may-13-2024/), we're requiring PostgreSQL 16 as of May 13.

# Solution

Update our documentation and Docker test images to reference v16.

# Testing

Ran the automated tests, plus have been using PostgreSQL v16 locally without issues for quite some time.

### CODE REVIEW

**Comment by mwiencek** [comment]
@brainzbot test this please

**Comment by mwiencek** [comment]
@brainzbot test this please

File: admin/InitDb.pl [hunk 1]
```diff
@@ -49,8 +49,8 @@ sub RequireMinimumPostgreSQLVersion
 
     my $version = $sql->select_single_value(q{SELECT current_setting('server_version_num')});
 
-    if ($version < 12000) {
-        die 'MusicBrainz requires PostgreSQL 12 or later';
+    if ($version < 160000) {
```
Comment (Kerollmops) [inline admin/InitDb.pl:52]:
```suggestion
    if ($version < 16000) {
```

Comment (mwiencek) [reply admin/InitDb.pl:52]: Thanks, I intended to leave a comment here but forgot. The previous check was incorrect; they increased the length of the number when their versioning system changed with major release 10. Here you can see what numbers are returned for v12, 15, and 16 respectively:

```
psql (12.16 (Ubuntu 12.16-1.pgdg22.04+1))
Type "help" for help.

musicbrainz_db=> SELECT current_setting('server_version_num');
 current_setting
-----------------
 120016
(1 row)
```

```
psql (15.6 (Homebrew), server 15.5 (Homebrew))
Type "help" for help.

musicbrainz_db=> SELECT current_setting('server_version_num');
 current_setting
-----------------
 150005
(1 row)
```

```
psql (16.1 (Debian 16.1-1.pgdg120+1))
Type "help" for help.

musicbrainz_json_dump=> SELECT current_setting('server_version_num');
 current_setting
-----------------
 160001
(1 row)
```

> Surprised this didn't cause any issues with our tests, don't we test InitDb?

Yes, but we don't test this particular condition (nor multiple versions of PostgreSQL).

**Comment by mwiencek** [comment]
> Tests pass - looking at https://www.postgresql.org/about/featurematrix/ I don't see any obvious places where we should refactor to use a much better new solution but you might, are there any?

The one feature I saw as immediately useful is the new `MERGE` statement, but it would take some time to identify situations where we could make use of it. We certainly have many such cases though.

**Comment by mwiencek** [comment]
@brainzbot test this please

**Comment by mwiencek** [comment]
@brainzbot test this please

**Comment by mwiencek** [comment]
@brainzbot test this please

**Pull request merged**